### ISSUE TITLE
docker -f deprecation -- now removed

### ISSUE THREAD

**Issue by mosieurlaurin** [mitchellh/packer#3771]
Docker has deprecated "-f" argument from tagging since v1.10.0
It has been removed in v.1.12.0

See https://docs.docker.com/engine/deprecated/#/f-flag-on-docker-tag

docker-tag post-processor returns the following when using packer with docker 1.12+:
`Stderr: unknown shorthand flag: 'f' in -f`

**Comment by dfarrell07** [comment]
[Here are some examples](http://pastebin.com/TMhU7T0x) of me hitting this with Docker 1.12.0 for the OpenDaylight Packer builds.

**Comment by dfarrell07** [comment]
https://github.com/docker/docker/pull/23090 is the PR that removed the flag and broke Packer. https://github.com/docker/docker/issues/24494 was a discussion about backwards compatibility, but it didn't seem to consider tools that use `tag -f` in an automated way, like Packer.

This consistently breaks tagging Docker images with Packer 0.10.1 (latest) and Docker 1.12.0 (latest). I tried downgrading Docker to the previous version, 1.11.2.

```
$ sudo systemctl stop docker
$ sudo dnf remove -y docker-engine docker-engine-selinux
# Note that I'm forcing F23 repos, as F24 only have 1.12.0
$ cat /etc/yum.repos.d/docker.repo
[dockerrepo]
name=Docker Repository
baseurl=https://yum.dockerproject.org/repo/main/fedora/23
enabled=1
gpgcheck=1
gpgkey=https://yum.dockerproject.org/gpg
$ sudo dnf install -y docker-engine-1.11.2-1.fc23 docker-engine-selinux-1.11.2-1.fc23
$ docker --version                                                                                                      
Docker version 1.11.2, build b9f10c9
```

But I run into https://github.com/docker/docker/issues/23981#issuecomment-230163925.

```
$ sudo systemctl start docker
A dependency job for docker.service failed. See 'journalctl -xe' for details.
$ sudo journalctl -xe
Aug 10 18:41:07 localhost.localdomain audit[1]: AVC avc:  denied  { create } for  pid=1 comm="systemd" scontext=system_u:system_r:init_t:s0 tcontext=system_u:system_r:unconfined_service_t:s0 tclass=unix_stream_socket permissive=0
Aug 10 18:41:07 localhost.localdomain systemd[1]: docker.socket: Failed to listen on sockets: Permission denied
```

[Full error logs](http://pastebin.com/FmFAyQCq). I tried https://github.com/docker/docker/issues/23981#issuecomment-231094489 as a workaround, basically an exception in an SELinux rule. That fixed the issue it targeted (1SELinux is preventing systemd from create access on the unix_stream_socket Unknown1), but not the one about docker.socket. With no more time to debug this, I set SELinux to permissive, which seems to work.

```
$ sudo setenforce 0
$ sudo systemctl start docker
$ sudo systemctl is-active docker                                                                        
active
$ docker run hello-world
# Works
$ docker --version                                                                                       
Docker version 1.11.2, build b9f10c9
```

[OpenDaylight's Packer builds](https://github.com/opendaylight/integration-packaging/tree/master/packer) also work.

```
$ packer build -var-file=vars/opendaylight-4.2.0.json -var-file=vars/fedora-23.json templates/docker.json
# Works
```

So I guess one work-around is to disable SELinux.

**Comment by thaJeztah** [comment]
Removing the `-f` / `--force` flag would make it work for all of docker 1.10.x, 1.11.x and 1.12.x (`--force` was the default starting with docker 1.10), perhaps that's the best option?

**Comment by oktopuz** [comment]
Right now the Feature Deprecation Policy says 'will remain in Docker for, usually, at least 3 releases'.

The downside of the -f removal were discussed several times before:

- https://github.com/docker/docker/pull/18350
- https://github.com/docker/docker/issues/24494
- https://github.com/docker/docker/pull/24721

Seriously why it is '[Not LGTM](https://github.com/docker/docker/pull/24721)' for keeping -f in 1.12.0 and be consistent with the current deprecation policy, while is it okay to [un-deprecate -c](https://github.com/docker/docker/pull/22621) for the same reason "is still widely used"?

**Comment by aloukissas** [comment]
If I read this correctly, docker-tag doesn't work with current packer if using docker 1.12 and there is no plan to fix it?

**Comment by rickard-von-essen** [comment]
@aloukissas the plan is:
1) Someone contribute a fix. 
2) Someone test it. 
3) Someone merge it. 

That is how most FOSS works *). 

*) Unless you are on a enterprise contract, then you should contact support.

**Comment by aloukissas** [comment]
@rickard-von-essen the fix is a couple of source code removals in the docker driver file AFAICT (highly unfamiliar with the packer code though). I'll fork and test it out manually - unsure of what the acceptance criteria in this project are (automated testing per PR, perhaps?)

**Comment by aloukissas** [comment]
In the meantime, is there perhaps a way to disable this in the release bits? One needs to install a year-old release of docker engine for packer to work :/

**Comment by rickard-von-essen** [comment]
Whats wrong with using `"force": false`? See [Docs: packer-tag - force](https://www.packer.io/docs/post-processors/docker-tag.html#force)

**Comment by dfarrell07** [comment]
> What's wrong with using `"force": false`?

That works. A [work-around like this](https://git.opendaylight.org/gerrit/#/c/47904/) gets Docker 1.12.3 working with Packer 0.10.1.

Thanks!

**Comment by mwhooker** [comment]
Should be fixed as part of #4186. going to track there

and looks like there's a viable workaround (don't use -f) for now
